File: ruby-neo4j-driver/connecting_paths.rb
require './neo4j_init'

# Get session
session = Neo4jInit.getSession("../config.yml")

# Connecting paths
connecting_paths_query = <<QUERY
  MATCH path = shortestPath((p1:Person)-[:KNOWS*..6]-(p2:Person))
  WHERE p1.name = {name1} AND p2.name = {name2}
  RETURN nodes(path) AS nodes
QUERY

response = session.query(connecting_paths_query, name1: 'Joe', name2: 'Billy')
puts response.first.nodes.map {|node| node.properties[:name] }
puts '---------------------'

File: elixir-neo4j-driver/social/connecting_paths.exs
[[{'userName', username}, {'password', password}, {'host', host}, _ , _, _]] = :yamerl_constr.file("../../config.yml")
{:ok, _neo} = Bolt.Sips.start_link(url: "bolt://#{username}:#{password}@#{host}")
conn = Bolt.Sips.conn()

# common friends
params = %{name1: "Joe", name2: "Billy"}
common_friends_query = """
  MATCH path = shortestPath((p1:Person)-[:KNOWS*..6]-(p2:Person))
  WHERE p1.name = {name1} AND p2.name = {name2}
  RETURN nodes(path) AS nodes
"""

Bolt.Sips.query!(conn, common_friends_query, params).results |> IO.inspect
